src/libponyc/codegen/genfun.c

```diff
@@ -97,8 +97,14 @@ static void make_signature(compile_t* c, reach_type_t* t,
   size_t tparam_size = count * sizeof(LLVMTypeRef);
 
   if(message_type)
+  {
     tparam_size += tparam_size + (2 * sizeof(LLVMTypeRef));
 
+    // must be at least the size of 3 or else we get a heap buffer overflow
+    if((2 * sizeof(LLVMTypeRef)) == tparam_size)
+      tparam_size += tparam_size + sizeof(LLVMTypeRef);
```

Comment:
Here's an alternative suggested fix for the address sanitizer issue here that is hopefully less confusing:

```suggestion
    if (tparam_size == 0)
      tparam_size = sizeof(LLVMTypeRef));
    tparam_size += tparam_size + (2 * sizeof(LLVMTypeRef));
```